ui/selectors/selectors.js

```diff
@@ -3579,11 +3579,14 @@ export function getOrderedConnectedAccountsForActiveTab(state) {
     // eslint-disable-next-line camelcase
     permissionHistory[activeTab.origin]?.eth_accounts?.accounts;
   const orderedAccounts = getMetaMaskAccountsOrdered(state);
-  const connectedAccounts = getPermittedEVMAccountsForCurrentTab(state);
+  const connectedAccounts = getAllPermittedAccountsForCurrentTab(state);
+  const connectedAccountsAddresses = connectedAccounts.map((acc) => {
+    const split = acc.split(':');
+    return split[split.length - 1];
+  });
```

Comment:
we should use `parseCaipAccountId` here instead.

This would looks something like
```suggestion
  const connectedAccounts = getAllPermittedAccountsForCurrentTab(state);
  const connectedAccountsAddresses = connectedAccounts.map((caipAccountId) => {
   const { address } = parseCaipAccountId(caipAccountId)
   return address
  });
```